core/src/main/scala/org/http4s/parser/SimpleHeaders.scala

```diff
@@ -23,11 +23,24 @@ import headers._
 import java.net.InetAddress
 import org.http4s.util.CaseInsensitiveString._
 
+import scalaz.NonEmptyList
+
 /**
  * parser rules for all headers that can be parsed with one simple rule
  */
 private[parser] trait SimpleHeaders { self: HttpParser =>
 
+  def ALLOW(value: String): ParseResult[Allow] = {
+    new Http4sHeaderParser[Allow](value) {
+      def entry = rule {
+        oneOrMore(Token).separatedBy(ListSep) ~ EOL ~>  { ts: Seq[String] =>
+          val ms = ts.map(Method.fromString(_).getOrElse(sys.error("Impossible. Please file a bug report.")))
```

Comment:
Not impossible, in case someone tried to put a non-token character in there.  But really, really, really weird.